Application/EdFi.Ods.Common/Providers/Queries/Criteria/IdentificationCodeAggregateQueryCriteriaApplicator.cs

```diff
@@ -0,0 +1,134 @@
+
+                if (property.IsDescriptorUsage && queryParameterValue == null)
+                {
+                    // Descriptor did not match any value -- criteria should exclude all entries
+                    // Since no additional criteria need to be applied, exit the loop
+                    queryBuilder.WhereRaw("1 = 0");
+                    break;
+                }
+
+                queryBuilder.Where($"{IdentificationCodeTableAlias}.{property.PropertyName}", queryParameterValue);
+            }
+        }
+        
+        object GetQueryParameterValueForProperty(EntityProperty property, string parameterValue)
+        {
+            if (!property.IsDescriptorUsage)
+            {
+                return parameterValue;
+            }
+
+            var lookupId = _descriptorResolver.GetDescriptorId(property.DescriptorEntity.Name, parameterValue);
+
+            return lookupId == 0
+                ? null
+                : lookupId;
+        }
+    }
+
+    private Join CreateIdentificationCodeTableJoin(EntityProperty entityIdentificationCodeProperty, string alias)
+    {
+        var join = new Join(
+            $"{entityIdentificationCodeProperty.Entity.Schema}.{entityIdentificationCodeProperty.Entity.TableName(_databaseEngine)}"
+                .Alias(IdentificationCodeTableAlias));
+
+        foreach (var entityIdentificationCodePropertyColumnName in GetIdentificationCodeEntityTableJoinColumnNames(
+                     entityIdentificationCodeProperty.Entity, _databaseEngine))
+        {
+            join.On($"{alias}.{entityIdentificationCodePropertyColumnName}", $"{IdentificationCodeTableAlias}.{entityIdentificationCodePropertyColumnName}");
+        }
+
+        return join;
+
+        IEnumerable<string> GetIdentificationCodeEntityTableJoinColumnNames(Entity identificationCodeEntity,
+            DatabaseEngine databaseEngine)
+        {
+            return identificationCodeEntity.Identifier.Properties
+                .Where(p => p.IsFromParent && p.IsIdentifying)
+                .Select(p => p.ColumnName(databaseEngine, p.PropertyName));
+        }
+    }
+
+    private IDictionary<string, EntityProperty> GetIdentificationCodeParameters(EntityProperty entityIdentificationCodeProperty)
+    {
+        return entityIdentificationCodeProperty.Entity.Properties
+            .Where(IsQueryableIdentificationCodeProperty)
+            .ToDictionary(GetParameterNameForIdentificationCodeProperty, StringComparer.OrdinalIgnoreCase);
+
+        string GetParameterNameForIdentificationCodeProperty(EntityProperty property)
+        {
+            return property.DescriptorEntity?.Name ?? property.PropertyName;
+        }
+        
+        bool IsQueryableIdentificationCodeProperty(EntityProperty property)
+        {
+            return property.PropertyName.Equals("IdentificationCode") ||
+                   !(property.IsFromParent || property.IsAlreadyDefinedInCSharpEntityBaseClasses());
```

Comment:
`IsAlreadyDefinedInCSharpEntityBaseClasses` is currently unused by the API project and probably should only be available for use by CodeGen. It looks like `IsPredefinedProperty` is in use and accomplishes the same thing. Let's use that.